intercept/chatcompletions/base.go

```diff
@@ -0,0 +1,249 @@
+package chatcompletions
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strings"
+
+	"github.com/coder/aibridge/config"
+	aibcontext "github.com/coder/aibridge/context"
+	"github.com/coder/aibridge/mcp"
+	"github.com/coder/aibridge/recorder"
+	"github.com/coder/aibridge/tracing"
+	"github.com/google/uuid"
+	"github.com/openai/openai-go/v2"
+	"github.com/openai/openai-go/v2/option"
+	"github.com/openai/openai-go/v2/shared"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+
+	"cdr.dev/slog"
+)
+
+type InterceptionBase struct {
```

Comment:
We need to be _very_ deliberate about only exporting types & funcs that could conceivably be used by users of this library.

I can't see a reason why it's exported, since it only has a package-local role.

_Remember: exported types & funcs increase the API surface, which means a larger maintenance burden for us going forward._